streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueStore.java

```diff
@@ -76,6 +76,8 @@ public void init(final ProcessorContext context,
                 false
             );
             // register the store
+            open = true;
```

Comment:
The store needs to be open for the processor to write to it when starting up. This actually matches the rocks store where it is opened right before registering instead of afterwards